### PR TITLE
ports brain tumor changes

### PR DESCRIPTION
**Pull request by KoboldCommando**
# About The Pull Request

This is a port of https://github.com/BeeStation/BeeStation-Hornet/pull/7172/

Makes Brain Tumor quirk less harsh, so that people would want to pick the quirk.

## Why It's Good For The Game

a quirk that nobody actually can help you is stupid. It's even worse than Blind trait in many situations.
This PR makes Brain Tumor penalty less harsh, so that you can earn your time more to find a help to treat your tumor.
Because it's less harsh, I added more downside in return.

## Changelog

:cl: EvilDragonfiend (ported by KoboldCommando)
balance: Brain Tumor will deal 0.1 brain damage at 80% chance. (equals 0.08 damage) previously it was 0.2 damage.
balance: Brain Tumor will not deal brain damage when the quirker has mannitol in their bloodstream.
balance: Mannitol is less effective for Brain Tumor quirker. It only heals 0.5 brain damage and special overdose thresholds(35u). on overdose, it will heal only 0.1 brain damage at 10% chance. (equals 0.01 damage) Don't overdose them.
balance: Brain Tumor quirker now starts with 5 pills of 30u mannitol. Each pill keeps you for 150 seconds, in total 750 seconds.
add: Brain Tumor will warn you when you're about to be at a mild trauma threshold. It will notice you when your brain damage is at near zero.
add: Brain Tumor quirker will have a +0 mood event when they eat mannitol. the message will persist for 30~60 seconds longer even if mannitol doesn't exist in your bloodstream.
/:cl:

### CODE REVIEW

**Comment by EvilDragonfiend** [comment]
It is not Bacon's PR lol

**Comment by KoboldCommando** [comment]
I may be slightly out of it right now, thank you!

File: code/datums/traits/negative.dm [hunk 1]
```diff
@@ -53,15 +53,27 @@
...
 	process = TRUE
 	var/where = "at your feet"
+	var/notified = FALSE
 
 /datum/quirk/brainproblems/on_process(delta_time)
-	quirk_holder.adjustOrganLoss(ORGAN_SLOT_BRAIN, 0.2 * delta_time)
+	if(!quirk_holder.reagents.has_reagent(/datum/reagent/medicine/mannitol))
+		if(prob(80))
+			quirk_holder.adjustOrganLoss(ORGAN_SLOT_BRAIN, 0.1 * delta_time)
+	var/obj/item/organ/brain/B = quirk_holder.getorgan(/obj/item/organ/brain)
```
Comment (nednaZ) [inline code/datums/traits/negative.dm:69]: One letter variable
```suggestion
	var/obj/item/organ/brain/affected_brain = quirk_holder.getorgan(/obj/item/organ/brain)
```

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 2]
```diff
@@ -843,8 +843,25 @@
...
+	else // to ordinary people
+		C.adjustOrganLoss(ORGAN_SLOT_BRAIN, -2*REM)
+	..()
+
+/datum/reagent/medicine/mannitol/overdose_process(mob/living/carbon/C)
+	if(HAS_TRAIT(C, TRAIT_BRAIN_TUMOR))
+		if(prob(10))
+			C.adjustOrganLoss(ORGAN_SLOT_BRAIN, -0.1*REM)
+	..()
 	..()
```
Comment (nednaZ) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:865]: Double parent call

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 3]
```diff
@@ -843,8 +843,25 @@
...
+		SEND_SIGNAL(C, COMSIG_ADD_MOOD_EVENT, "brain_tumor", /datum/mood_event/brain_tumor_mannitol)
+		if(!overdosed)
+			C.adjustOrganLoss(ORGAN_SLOT_BRAIN, -0.5*REM)
+	else // to ordinary people
+		C.adjustOrganLoss(ORGAN_SLOT_BRAIN, -2*REM)
+	..()
+
+/datum/reagent/medicine/mannitol/overdose_process(mob/living/carbon/C)
+	if(HAS_TRAIT(C, TRAIT_BRAIN_TUMOR))
+		if(prob(10))
```
Comment (nednaZ) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:862]: You can only OD if you have the brain tumor trait, so you can clear the extra check

**Review by nednaZ — CHANGES_REQUESTED** [review]
Good changes for making this quirk a bit more bearable.
Just needs the minor cleanup.

**Comment by KoboldCommando** [comment]
All changes applied, thank you!

**Comment by nednaZ** [comment]
Looks like you changed the name of the variable, but not the uses of the variable.

**Comment by KoboldCommando** [comment]
I had missed one final use of the variable, sorry, fixed again!

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 3 above)
Comment (EvilDragonfiend) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: That was intentional. Only tumor people have OD threshold to make their own downside.

Comment (EvilDragonfiend) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: It was a safe check when you are OD'ed with mannitol for some reason (admin, or new feature.)
Although it doesn't do much.

**Review by nednaZ — APPROVED** [review]
Looks good to me.

**Pull request merged**